app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/home/birthday/BirthdayViewModel.kt

```diff
@@ -33,68 +35,147 @@ class BirthdayViewModel(
     }.stateIn(viewModelScope, SharingStarted.Lazily, false)
 
     fun setName(name: String) {
         _name.update { name }
     }
 
-    fun setImage(image: Uri?) {
-        _imageUri.update { image }
+    fun setImage(image: ByteArray?) {
+        _imageByteArray.update { image }
     }
 
     fun setRelationship(relationship: RelationshipEnum?) {
         _relationship.update { relationship }
     }
 
-    fun setDate(date: Long?) {
-        _date.update { if (date != null) Date(date) else null }
-        _dateTitle.update { formatter.format(Date(date ?: Calendar.getInstance().timeInMillis)) }
+    fun setDate(dateUtc: String) {
+        _dateUtc.update { dateUtc }
+        _dateDayMonthYear.update { dateUtc.fromUtcToDayMonthYearDate() }
     }
 
     fun setCreateBirthdayErrorFalse() {
         _createBirthdayError.value = false
     }
 
-    fun createBirthday(navigateToHomeScreen: () -> Unit) {
+    fun setUpdateBirthdayErrorFalse() {
+        _updateBirthdayError.value = false
+    }
+
+    fun setDeleteBirthdayErrorFalse() {
+        _deleteBirthdayError.value = false
+    }
+
+    fun setBirthday(birthday: Birthday?) {
+        _editModeBirthday.update { birthday }
+        if (birthday != null) {
+            setName(birthday.name)
+            setImage(birthday.image)
+            setRelationship(birthday.relation)
+            setDate(birthday.dateUtc)
+        } else {
+            setName("")
+            setImage(null)
+            setRelationship(null)
+            _dateUtc.update { Calendar.getInstance().timeInMillis.fromMillisToUtcDate() }
+            _dateDayMonthYear.update { "__.__.____" }
+        }
+    }
+
+    fun createBirthday(
+        navigateToHomeScreen: () -> Unit,
+        onCompletion: () -> Unit,
+    ) {
         viewModelScope.launch(Dispatchers.IO) {
-            val imageUri = imageUri.value
+            val imageByteArray = imageByteArray.value
             birthdayRepository.createBirthday(
                 createBirthday = CreateBirthdayEntity(
                     name = _name.value,
-                    imageBase64 = imageUri?.uriToBase64(context = getApplication<Application>().applicationContext),
+                    imageBase64 = imageByteArray?.uriToBase64(),
                     relation = getApplication<Application>().applicationContext.getString(
                         _relationship.value?.resId ?: RelationshipEnum.BEST_FRIEND.resId
                     ),
-                    date = _date.value.toString(),
+                    dateUtc = _dateUtc.value,
                 ),
             ).onEach {
                 it.onSuccess { navigateToHomeScreen() }
                 it.onFailure { _createBirthdayError.update { true } }
-            }.catch {
-                _createBirthdayError.update { true }
-            }.collect()
+            }.onCompletion { onCompletion() }
+                .catch { _createBirthdayError.update { true } }
+                .flowOn(Dispatchers.Main).collect()
```

Comment:
Don't need pass dispatcher Dispatchers.Main , it is already on Dispatchers.IO.. Please fix this in other places too.